### PR TITLE
fix comparing types and use isinstance

### PR DESCRIPTION
**Pull request by rgildein**
Replace types comparison with usage of `isinstance` function.

### CODE REVIEW

File: tests/unit/test_passwd.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def test_user_exists_by_uid_true(self, getpwuid):
 
     def test_user_exists_invalid_input(self):
         with self.assertRaises(TypeError):
-            passwd.user_exists(True)
+            print(passwd.user_exists(True))
```
Comment (benhoyt) [inline tests/unit/test_passwd.py:24]: Is this left-over debugging?

**Review by benhoyt — CHANGES_REQUESTED** [review]
Won't the patch versions need incrementing due to these changes?

**Comment by benhoyt** [comment]
Other than that, though, this looks good to me -- thanks!

**Comment by NucciTheBoss** [comment]
This pull request should fix the failing lint tests in https://github.com/canonical/operator-libs-linux/pull/106

File: tests/unit/test_passwd.py (hunk 1 above)
Comment (benhoyt) [inline tests/unit/test_passwd.py:24]:
```suggestion
            passwd.user_exists(True)
```

**Review by benhoyt — APPROVED** [review]

File: tests/unit/test_passwd.py (hunk 1 above)
Comment (rgildein) [reply tests/unit/test_passwd.py]: Yes, sorry for that

**Pull request merged**